app/src/main/java/ru/practicum/android/diploma/filter/data/storage/FilterStorageImpl.kt

```diff
@@ -34,26 +35,35 @@ class FilterStorageImpl(private val sharedPref: SharedPreferences, private val g
     }
 
     override suspend fun saveSalary(salary: Int?) {
-        val params = getParamsOrCreateFilter()
-        params.salary = salary
+        val params = getParamsOrCreateFilter().apply { this.salary = salary }
         updateField(params)
     }
 
     override suspend fun saveSalaryFlag(onlyWithSalary: Boolean?) {
-        val params = getParamsOrCreateFilter()
-        params.onlyWithSalary = onlyWithSalary
+        val params = getParamsOrCreateFilter().apply { this.onlyWithSalary = onlyWithSalary }
         updateField(params)
     }
 
     override suspend fun clearFilterSettings() {
         sharedPref.edit().remove(FILTER_PARAMETERS).apply()
     }
 
+    override suspend fun restoreFilterSettings(filterParameters: FilterParameters?) {
+        val json = if(filterParameters != null ) gson.toJson(filterParameters.toDto()) else null
+        sharedPref.edit()
+            .putString(FILTER_PARAMETERS, json)
+            .apply()
+    }
+
     override suspend fun getParams(): FilterParameters? {
         val json = sharedPref.getString(FILTER_PARAMETERS, null) ?: return null
         val filterParametersDto =
             gson.fromJson(json, FilterParametersDto::class.java)
-        return filterParametersDto.toDomain()
+        return if (filterParametersDto.country == null && filterParametersDto.area == null && filterParametersDto.industry == null && filterParametersDto.salary == null && filterParametersDto.onlyWithSalary == null) {
```

Comment:
Слишком длинная конструкция, разбейте поменьше
```suggestion
        return if (filterParametersDto.country == null && 
        filterParametersDto.area == null && 
        filterParametersDto.industry == null && 
        filterParametersDto.salary == null && 
        filterParametersDto.onlyWithSalary == null) {
```

Reply:
исправила в ветке 200